packages/platform-android/src/commands/runAndroid/runOnAllDevices.ts

```diff
@@ -52,6 +42,10 @@ async function runOnAllDevices(
     }
   }
 
+
+  await buildApk(args, gradlew, androidProject);
+  await installApk(args, gradlew, adbPath, devices, androidProject);
+
   try {
```

Comment:
I believe this code is already moved out to `buildApk` and `installApk` respectively, e.g. `args.port` and `createInstallError`. I am going to go ahead and remove it.